rust/lance-index/src/scalar/inverted/tokenizer.rs

```diff
@@ -141,9 +145,72 @@ fn build_base_tokenizer_builder(name: &str) -> Result<tantivy::tokenizer::TextAn
             tantivy::tokenizer::RawTokenizer::default(),
         )
         .dynamic()),
+        #[cfg(feature = "tokenizer-lindera")]
+        s if s.starts_with("lindera/") => {
+            return build_lindera_tokenizer_builder(s);
+        }
         _ => Err(Error::invalid_input(
             format!("unknown base tokenizer {}", name),
             location!(),
         )),
     }
 }
+
+lazy_static::lazy_static! {
+    pub static ref LANCE_TOKENIZER_HOME: Option<PathBuf> = LANCE_HOME.as_ref().map(|p| p.join("tokenizers"));
+}
+
+#[cfg(feature = "tokenizer-lindera")]
+fn build_lindera_tokenizer_builder(dic: &str) -> Result<tantivy::tokenizer::TextAnalyzerBuilder> {
+    use std::{fs::File, io::BufReader};
+
+    use lindera::{
+        dictionary::{
+            load_dictionary_from_path, load_user_dictionary_from_config, UserDictionaryConfig,
+        },
+        mode::Mode,
+        segmenter::Segmenter,
+    };
+    use lindera_tantivy::tokenizer::LinderaTokenizer;
+    use serde_json::from_reader;
+
+    match LANCE_TOKENIZER_HOME.as_ref() {
+        Some(p) => {
+            let dic_dir = p.join(dic);
+            let main_dir = dic_dir.join("main");
+            let user_config_path = dic_dir.join("user_config.json");
```

Comment:
What's the schema of this JSON file?

Reply:
it's https://github.com/lindera/lindera/blob/0a92b32ca7e90e3b1959d17ca745057e7d3c8314/lindera/src/dictionary.rs#L183. includes path and kind at this time. I can remove this file. if the path contains kind name. convention over configuration.

Reply:
I have updated implementation, with a clear configuration schema